### PR TITLE
[INFRA] Add k8s pipeline

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add a github pipeline to run the tests of k8s

### Why are the changes needed?
It can help us to verify the k8s patch

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
No

### CODE REVIEW

File: .github/workflows/parallel.yml [hunk 1]
```diff
@@ -76,6 +77,10 @@ jobs:
         java-version: ${{ inputs.java-version }}
         distribution: ${{ inputs.jdk-distro }}
         cache: maven
+    - name: Set up Go 1.16
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.16
```
[the remark this answers is not in the document]
Comment (jerqi) [reply .github/workflows/parallel.yml:83]: The operator is implemented by version 1.16 now. If necessary, we can upgrade it.

**Comment by jerqi** [comment]
Could you help me review this patch? @zuston @kaijchen

**Review by zuston — APPROVED** [review]
LGTM

File: .github/workflows/parallel.yml [hunk 2]
```diff
@@ -76,6 +77,10 @@ jobs:
         java-version: ${{ inputs.java-version }}
         distribution: ${{ inputs.jdk-distro }}
         cache: maven
+    - name: Set up Go 1.16
+      uses: actions/setup-go@v3
```
Comment (kaijchen) [inline .github/workflows/parallel.yml:81]: We can set `if: ${{ matrix.profile ==  kubernetes}}` here, although it is not required.

Comment (jerqi) [reply .github/workflows/parallel.yml:81]: Could you raise a good first issue about this optimization?

**Pull request merged**